Enhance event name for documentation link and update tracking document.

ianlin opened the pull request:
### Changes proposed in this Pull Request:

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

Closes #1497.

This PR removes the `eventName` prop from the component that uses [AppDocumentationLink](https://github.com/woocommerce/google-listings-and-ads/blob/cbc4a9d3b66279eeee8225d1ae0b97cdc1216d29/js/src/components/app-documentation-link/index.js), to make sure the eventName will always be `gla_documentation_link_click` defined in `AppDocumentationLink`.

This PR also add several missing event JSDocs for the event `gla_documentation_link_click`, as well as refactor some components' default export definition so the tracking doc can be generated correctly. To be clear, replace

```js
export default function SomeComponent {
}
```

with

```js
const SomeComponent = () => {
};

export default SomeComponent;
```

So it will fix the problem that in [Tracking Doc](https://github.com/woocommerce/google-listings-and-ads/blob/c686f6ed423992e1627186bf49e208494027f4e7/src/Tracking/README.md) there are some component name generated as `exports`.

### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

#### Check event name is sent correctly

1. Open your console and run `localStorage.setItem('debug', 'wc-admin*')`.
2. Enable verbose mode in dev tool's console.
3. Modify [js/src/get-started-page/unsupported-notices/index.js](https://github.com/woocommerce/google-listings-and-ads/blob/c686f6ed423992e1627186bf49e208494027f4e7/js/src/get-started-page/unsupported-notices/index.js) locally so the unsupported notice can show up
    ```diff
   --- a/js/src/get-started-page/unsupported-notices/index.js
   +++ b/js/src/get-started-page/unsupported-notices/index.js
   @@ -119,8 +119,8 @@ export default function UnsupportedNotices() {
   
     return (
       <>
   -     { ! mcSupportedLanguage && <UnsupportedLanguage /> }
   -     { ! mcSupportedCountry && <UnsupportedCountry /> }
   +     { mcSupportedLanguage && <UnsupportedLanguage /> }
   +     { mcSupportedCountry && <UnsupportedCountry /> }
       </>
     );
    }
    ```
4. Modify [js/src/components/different-currency-notice.js](https://github.com/woocommerce/google-listings-and-ads/blob/c686f6ed423992e1627186bf49e208494027f4e7/js/src/components/different-currency-notice.js) locally so the different currency notice can show up
    ```diff
    --- a/js/src/components/different-currency-notice.js
    +++ b/js/src/components/different-currency-notice.js
    @@ -30,7 +30,7 @@ const DifferentCurrencyNotice = ( { context } ) => {
      if (
        ! googleAdsAccount ||
        googleAdsAccount.status !== 'connected' ||
    -   googleAdsAccount.currency === storeCurrency
    +   googleAdsAccount.currency !== storeCurrency
      ) {
        return null;
      }
    ```
5. Go to GLA's get started page: `https://domain.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fstart` 
6. Click on the two `Read more...` links on both notices, the event name should be `wcadmin_gla_documentation_link_click`
    - <img width="1278" alt="Screenshot 2022-05-12 at 18 05 21" src="https://user-images.githubusercontent.com/914406/168051092-e7f6952f-b8a4-4884-8fc6-0a61ffff50ca.png">
7. Connect Google MC account and go to GLA's dashboard and reports page
8. Click on the two `Read more` links on both notices, the event name should be `wcadmin_gla_documentation_link_click`
    - <img width="1703" alt="Screenshot 2022-05-12 at 18 16 26" src="https://user-images.githubusercontent.com/914406/168051487-7d13f4b1-a0dc-41a3-8a50-82cbf807e786.png">

#### Tracking Document

- Check the new tracking doc in the last commit, there shouldn't be `exports` as the component name anymore.
- There are several emitters under `gla_documentation_link_click` event as added by commit 0e7c950.

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with `(Fix|Add|Tweak|Update) - `, for example:
> Fix - I took care of something that wasn't working.
> Add - I added something new that's pretty cool.
> Tweak - I made a small change.
> Update - I made big changes to something that wasn't broken.

Leave the "Changelog entry" header in place completely empty, without any summary if no changelog entry is needed.
If you remove the "Changelog entry" header, the title of Pull Request will be used as the changelog entry.
-->
### Changelog entry

> Tweak - Enhance event name for documentation link and update tracking document.

ianlin committed:
Remove eventName from  AppDocumentationLink in UnsupportedLanguage and
UnsupportedCountry

ianlin committed:
Remove eventName from  AppDocumentationLink in DifferentCurrencyNotice

ianlin committed:
Add missing event JSDoc for gla_documentation_link_click

ianlin committed:
Makes tracking doc can generate the correct emitter names

ianlin committed:
Update tracking doc

puntope commented on js/src/get-started-page/faqs/index.js:
```diff
@@ -256,6 +256,13 @@ const faqItems = [
 
 /**
  * @fires gla_get_started_faq
```
💅  What about including here as well the id and action props?

puntope commented on js/src/setup-ads/ads-stepper/setup-accounts/free-ad-credit/index.js:
```diff
@@ -22,6 +22,7 @@ import CountryModal from './country-modal';
 
 /**
  * @fires gla_free_ad_credit_country_click
```
💅 
context prop missed in `gla_free_ad_credit_country_click`

puntope commented on js/src/setup-mc/setup-stepper/setup-accounts/faqs.js:
```diff
@@ -69,11 +69,14 @@ const faqItems = [
 
 /**
  * @fires gla_setup_mc_faq
```
💅  Missing id and action props.

💅  This could promote to a new refactor issue since `gla_setup_mc_faq` is kinda the same as `gla_get_started_faq` and other faqs. Could be interesting to do a refactor on this by using just `gla_faq` adding  `setup_mc` and `get_started` as props (context?)

puntope commented on src/Tracking/README.md:
```diff
@@ -198,6 +198,46 @@ When a documentation link is clicked.
 `href` | `string` | link's URL
 #### Emitters
 - [`AppDocumentationLink`](../../js/src/components/app-documentation-link/index.js#L29)
+- [`ContactInformation`](../../js/src/components/contact-information/index.js#L76) with `{ context: 'setup-mc-contact-information|settings-no-phone-number-notice|settings-no-store-address-notice', link_id: 'contact-information-read-more', href: 'https://docs.woocommerce.com/document/google-listings-and-ads/#contact-information' }`
+- [`DifferentCurrencyNotice`](../../js/src/components/different-currency-notice.js#L25) with `{ context: "dashboard|reports-products|reports-programs", link_id: "setting-up-currency", href: "https://support.google.com/google-ads/answer/9841530" }`
```
💅  To consider to create one line per context?

puntope commented:
> This PR removes the eventName prop from the component that uses [AppDocumentationLink](https://github.com/woocommerce/google-listings-and-ads/blob/cbc4a9d3b66279eeee8225d1ae0b97cdc1216d29/js/src/components/app-documentation-link/index.js), to make sure the eventName will always be gla_documentation_link_click defined in AppDocumentationLink.

❓ I don't see any mod on `AppDocumentationLink` avoiding that.

puntope reviewed (commented):
👍  Thanks for the changes @ianlin 

I left some comments and one question to clarify ❓  I think we are not solving the potential `AppDocumentationLink` event override.

ianlin committed:
Add event JSDoc for gla_get_started_faq event

ianlin committed:
Add props for gla_free_ad_credit_country_click event in event JSDoc

ianlin committed:
Combine events gla_setup_mc_faq and gla_get_started_faq as gla_faq

And add a `context` event prop to indicate which page it's in.

ianlin committed:
Update event JSDoc to make it one line per property

ianlin committed:
Put eventName after spreading `rest` to prevent it from being overridden

Since we want to make sure whoever uses AppDocumentationLink the
eventName should always be gla_documentation_link_click.

ianlin committed:
Update event tracking document

ianlin commented:
Thanks for the review @puntope! I've updated the codes based on your suggestion and it's ready for another round of review.

puntope reviewed (approved):
✅  LGTM

Thanks @ianlin
